diff --git a/widget/PuppetWidget.h b/widget/PuppetWidget.h
index e036af1e43c1..767bca967f67 100644
--- a/widget/PuppetWidget.h
+++ b/widget/PuppetWidget.h
@@ -302,8 +302,14 @@ class PuppetWidget final : public nsIWidget,
   void OnMemoryPressure(layers::MemoryPressureReason aWhy) override;
 
   void PerformHapticFeedback(mozilla::HapticFeedbackType aType) override;
+  [[nodiscard]] bool ShouldAdjustIMECacheForScroll();
+  void AdjustIMECacheForScroll(const CSSPoint& aNewScrollOffset);
 
  private:
+  CSSPoint mIMELastScrollOffset;
+
+  void RecordIMECacheScrollOffset();
+
   void Paint();
 
   nsresult RequestIMEToCommitComposition(bool aCancel);
