diff --git a/layout/base/nsCaret.cpp b/layout/base/nsCaret.cpp
index e61e4ba19c87..6a62c3984934 100644
--- a/layout/base/nsCaret.cpp
+++ b/layout/base/nsCaret.cpp
@@ -16,6 +16,7 @@
 #include "mozilla/PresShell.h"
 #include "mozilla/ScrollContainerFrame.h"
 #include "mozilla/StaticPrefs_bidi.h"
+#include "mozilla/TextControlElement.h"
 #include "mozilla/dom/CharacterDataBuffer.h"
 #include "mozilla/dom/Selection.h"
 #include "mozilla/gfx/2D.h"
@@ -474,6 +475,26 @@ nsIFrame* nsCaret::GetPaintGeometry(nsRect* aCaretRect, nsRect* aHookRect,
 
   nsIFrame* frame = data.mFrame;
   nsIFrame* unadjustedFrame = data.mUnadjustedFrame;
+#ifdef MOZ_WIDGET_UIKIT
+  // UIKit handles the caret and selection
+  if (!mReadOnly && unadjustedFrame->ContentIsEditable()) {
+    return nullptr;
+  }
+
+  nsIContent* content = unadjustedFrame->GetContent();
+  nsIContent* control =
+      content && content->IsInNativeAnonymousSubtree()
+          ? content->GetClosestNativeAnonymousSubtreeRootParentOrHost()
+          : content;
+
+  if (auto* textControl = TextControlElement::FromNodeOrNull(control)) {
+    if (textControl->IsSingleLineTextControlOrTextArea() &&
+        !textControl->IsDisabled()) {
+      return nullptr;
+    }
+  }
+#endif
+
   int32_t frameOffset(data.mOffsetInFrameContent);
   // Now we have a frame, check whether it's appropriate to show the caret here.
   // Note we need to check the unadjusted frame, otherwise consider the
