diff --git a/gfx/layers/apz/util/ChromeProcessController.cpp b/gfx/layers/apz/util/ChromeProcessController.cpp
index 5f1c230377be..2af823695b8c 100644
--- a/gfx/layers/apz/util/ChromeProcessController.cpp
+++ b/gfx/layers/apz/util/ChromeProcessController.cpp
@@ -197,9 +197,13 @@ void ChromeProcessController::HandleTap(
 
   switch (aType) {
     case TapType::eSingleTap: {
+#if defined(MOZ_WIDGET_UIKIT)
+      break;
+#else
       RefPtr<APZEventState> eventState(mAPZEventState);
       eventState->ProcessSingleTap(point, scale, aModifiers, 1, aInputBlockId);
       break;
+#endif
     }
     case TapType::eDoubleTap:
       MOZ_ASSERT(aDoubleTapToZoomMetrics);
