diff --git a/toolkit/components/extensions/child/ext-storage.js b/toolkit/components/extensions/child/ext-storage.js
index 5a9fb62957..1c5df2834b 100644
--- a/toolkit/components/extensions/child/ext-storage.js
+++ b/toolkit/components/extensions/child/ext-storage.js
@@ -8,6 +8,8 @@ ChromeUtils.defineESModuleGetters(this, {
   ExtensionStorage: "resource://gre/modules/ExtensionStorage.sys.mjs",
   ExtensionStorageIDB: "resource://gre/modules/ExtensionStorageIDB.sys.mjs",
   ExtensionTelemetry: "resource://gre/modules/ExtensionTelemetry.sys.mjs",
+  FullPageCaptureCompat:
+    "resource://gre/modules/FullPageCaptureCompat.sys.mjs",
 });

 // Wrap a storage operation in a TelemetryStopWatch.
@@ -151,6 +153,7 @@ this.storage = class extends ExtensionAPI {

   getAPI(context) {
     const { extension } = context;
+    FullPageCaptureCompat.applyPopupLayout(context);
     const serialize = ExtensionStorage.serializeForContext.bind(null, context);
     const deserialize = ExtensionStorage.deserializeForContext.bind(
       null,
