diff --git a/dom/events/IMEStateManager.cpp b/dom/events/IMEStateManager.cpp
index 97972ede68b0..b445788d4172 100644
--- a/dom/events/IMEStateManager.cpp
+++ b/dom/events/IMEStateManager.cpp
@@ -2189,6 +2189,34 @@ void IMEStateManager::SetIMEState(const IMEState& aState,
     GetAutocapitalize(aState, *focusedElement, context,
                       context.mAutocapitalize);
     context.mAutocorrect = GetAutocorrect(aState, *focusedElement, context);
+#ifdef XP_IOS
+    context.mHasEditableHost = focusedElement->IsEditable();
+    context.mSpellCheckingEnabled =
+        nsGenericHTMLElement::FromNode(focusedElement)->Spellcheck();
+
+    for (const Element* element = focusedElement; element;
+         element = element->GetParentElement()) {
+      if (!element->HasAttr(nsGkAtoms::writingsuggestions)) {
+        continue;
+      }
+
+      context.mWritingSuggestionsEnabled = !element->AttrValueIs(
+          kNameSpaceID_None, nsGkAtoms::writingsuggestions, nsGkAtoms::_false,
+          eIgnoreCase);
+      break;
+    }
+
+    if (auto* control = TextControlElement::FromNode(focusedElement)) {
+      if (control->IsSingleLineTextControlOrTextArea() &&
+          !control->IsDisabled()) {
+        context.mReadOnly = control->IsReadOnly();
+      }
+    }
+
+    if (HTMLInputElement* input = HTMLInputElement::FromNode(focusedElement)) {
+      context.mHasBeenPassword = input->HasBeenTypePassword();
+    }
+#endif
   }
 
   if (aAction.mCause == InputContextAction::CAUSE_UNKNOWN &&
