diff --git a/widget/uikit/GeckoViewSwiftSupport.h b/widget/uikit/GeckoViewSwiftSupport.h
index e68fca9134c0..7acd432aa05d 100644
--- a/widget/uikit/GeckoViewSwiftSupport.h
+++ b/widget/uikit/GeckoViewSwiftSupport.h
@@ -18,6 +18,7 @@
 #endif
 
 #import <Foundation/Foundation.h>
+#import <AVFoundation/AVSampleBufferDisplayLayer.h>
 #import <UIKit/UIKit.h>
 #import <xpc/xpc.h>
 
@@ -26,6 +27,9 @@
 @protocol SwiftGeckoViewRuntime <NSObject>
 - (id<SwiftEventDispatcher>)runtimeDispatcher;
 - (id<SwiftEventDispatcher>)dispatcherByName:(const char*)name;
+@optional
+- (void)childProcessDidStartWithPID:(int32_t)pid
+                        processType:(NSString*)processType;
 @end
 
 @protocol GeckoProcessExtension <NSObject>
@@ -56,6 +60,7 @@
 
 @protocol GeckoViewWindow <NSObject>
 - (UIView*)view;
+- (AVSampleBufferDisplayLayer*)pictureInPictureDisplayLayer;
 - (void)close;
 @end
 
