diff --git a/widget/uikit/TextInputHandler.mm b/widget/uikit/TextInputHandler.mm
index e8f05ad59f6f..81ce0e0ceded 100644
--- a/widget/uikit/TextInputHandler.mm
+++ b/widget/uikit/TextInputHandler.mm
@@ -20,6 +20,16 @@
 
 mozilla::LazyLogModule gIMELog("TextInputHandler");
 
+@interface ChildView : UIView
+
+@end
+
+@interface ChildView (TextInputNotifications)
+
+- (void)selectionGeometryChanged;
+
+@end
+
 namespace mozilla::widget {
 
 NS_IMPL_ISUPPORTS(TextInputHandler, TextEventDispatcherListener,
@@ -30,11 +40,28 @@
 
 nsresult TextInputHandler::NotifyIME(TextEventDispatcher* aTextEventDispatcher,
                                      const IMENotification& aNotification) {
+  if (Destroyed() || aTextEventDispatcher != mDispatcher) {
+    return NS_OK;
+  }
+
+  switch (aNotification.mMessage) {
+    case NOTIFY_IME_OF_POSITION_CHANGE: {
+      ChildView* view =
+          static_cast<ChildView*>(mWidget->GetNativeData(NS_NATIVE_WIDGET));
+      if (view) {
+        [view selectionGeometryChanged];
+      }
+      break;
+    }
+    default:
+      break;
+  }
+
   return NS_OK;
 }
 
 IMENotificationRequests TextInputHandler::GetIMENotificationRequests() {
-  return IMENotificationRequests{};
+  return IMENotificationRequests{IMENotificationRequest::PositionChange};
 }
 
 void TextInputHandler::OnRemovedFrom(
