diff --git a/widget/uikit/UIKitUtils.mm b/widget/uikit/UIKitUtils.mm
index 3c259b196fdd..5c8b81a47f08 100644
--- a/widget/uikit/UIKitUtils.mm
+++ b/widget/uikit/UIKitUtils.mm
@@ -11,7 +11,7 @@
   if (aContext.mHTMLInputMode.EqualsLiteral("email")) {
     return UIKeyboardTypeEmailAddress;
   }
-  if (aContext.mHTMLInputMode.EqualsLiteral("deciaml")) {
+  if (aContext.mHTMLInputMode.EqualsLiteral("decimal")) {
     return UIKeyboardTypeDecimalPad;
   }
   if (aContext.mHTMLInputMode.EqualsLiteral("numeric")) {
@@ -23,6 +23,9 @@
   if (aContext.mHTMLInputMode.EqualsLiteral("tel")) {
     return UIKeyboardTypePhonePad;
   }
+  if (aContext.mHTMLInputMode.EqualsLiteral("text")) {
+    return UIKeyboardTypeDefault;
+  }
   if (aContext.mHTMLInputMode.EqualsLiteral("url")) {
     return UIKeyboardTypeURL;
   }
@@ -31,7 +34,7 @@
     return UIKeyboardTypeEmailAddress;
   }
   if (aContext.mHTMLInputType.EqualsLiteral("number")) {
-    return UIKeyboardTypeNumberPad;
+    return UIKeyboardTypeNumbersAndPunctuation;
   }
   if (aContext.mHTMLInputType.EqualsLiteral("tel")) {
     return UIKeyboardTypePhonePad;
@@ -88,6 +91,12 @@
 // static
 UITextAutocorrectionType UIKitUtils::GetUITextAutocorrectionType(
     const InputContext& aContext) {
+  if (aContext.IsPasswordEditor() ||
+      aContext.mHTMLInputType.EqualsLiteral("email") ||
+      aContext.mHTMLInputType.EqualsLiteral("url")) {
+    return UITextAutocorrectionTypeNo;
+  }
+
   return aContext.mAutocorrect ? UITextAutocorrectionTypeYes
                                : UITextAutocorrectionTypeNo;
 }
